src/libraries/System.Transactions.Local/src/System/Transactions/DtcProxyShim/DtcProxyShimFactory.cs

```diff
@@ -117,6 +116,12 @@ internal DtcProxyShimFactory(EventWaitHandle notificationEventHandle)
             resourceManagerShim = rmShim;
             _transactionDispenser = localDispenser;
             whereabouts = tmpWhereabouts;
+
+            [UnconditionalSuppressMessage("Trimming", "IL2050", Justification = "Leave me alone")]
```

Comment:
@AaronRobinsonMSFT is this a leftover from your code review ;-) (it sounds like you)
In any case @jkurdek could you please change the Justification to something more professional - like "The PInvoke has object/interface typed parameters which are potentially trim incompatible, but in this case they're OK".

@AaronRobinsonMSFT - I don't know the details of the implicit COM interop , but I'm not actually sure this IS OK - There are two parameters which are potentially problematic:
`pvConfigParams` (typed as `object`) - but we're passing in `null`, so there's no trimming problem.
`ppvObject` which is declared as `[MarshalAs(UnmanagedType.Interface)] out ITransactionDispenser ppvObject` - my understanding is that the interop will take the managed definition of `ITransactionDespenser` and assume the vtable coming from native matches the managed definition. With trimming, that means we have to guarantee that we preserve the entire `ITransactionDespenser` as-is. Additionally the code will try to cast this to couple of other interface types (`ITmNodeName`, `ITransactionImportWhereabouts` and `IResourceManagerFactory2`) all of which will need the same treatment I guess.

Isn't this something which should use `ComWrappers` to be trim-compatible? I guess that is covered in https://github.com/dotnet/runtime/issues/75031 ?

Reply:
> is this a leftover from your code review ;-) (it sounds like you)

100 % me, apologies. Please update as appropriate.

> I don't know the details of the implicit COM interop

I think there was discussion that the Windows version of Transactions wasn't trim-compatible at all. However, how we express that isn't clear to me and at the time I was simply trying to get it to "leave me alone". Your analysis makes sense to me as far as why it may not be okay.

> Isn't this something which should use ComWrappers to be trim-compatible? I guess that is covered in https://github.com/dotnet/runtime/issues/75031 ?

From my conversations this particular part is a legacy or at the very least non-forward looking feature set that will not have a lot of investment in the future. It is unclear if `ComWrappers` will be worth the long term effort but perhaps #75031 is meant to track that conversation. @roji is the product owner here and can help guide us toward the correct solution.

Reply:
Ah sorry, the unprofessional comment was my bad - @AaronRobinsonMSFT just proposed a draft PR with this, and I was supposed to change the text. Definitely go ahead and update as appropriate.

As @AaronRobinsonMSFT wrote, System.Transactions generally isn't a forward-looking components; the reason we ported the code containing the COM interop is to unblock .NET Framework users depending on distributed transactions and allow them to port to .NET. I very much doubt we'll invest more in this in the future; if we see a real need for a cross-platform API, we'll be implementing new support for that (#71769), but that likely wouldn't involve direct COM interop in any way.

I'll post a comment in #75031 clarifying the situation.